### PR TITLE
Add CentOS7 dockerfile

### PR DESCRIPTION
**Pull request by discogestalt**
Add support for rpm building to CentOS 8 dockerfile

### CODE REVIEW

File: swift-ci/master/centos/8/Dockerfile [hunk 1]
```diff
@@ -30,6 +30,7 @@ RUN yum install --enablerepo=PowerTools -y \
     python3-six           \
     python3-pexpect       \
     platform-python-devel \
+    rpm-build.x86_64      \
```
Comment (tomerd) [inline swift-ci/master/centos/8/Dockerfile]: hi @discogestalt we should remove  rpm-build package from these docker images - we do not intend to create RPMs using these docker images want to keep the list minimal

File: swift-ci/master/centos/7/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	python-pygments   \
+	python36-pexpect  \
+	PyYAML            \
+	rpm-build.x86_64  \
+	rsync             \
+	sclo-git25-git    \
+	sqlite-devel      \
+	swig3             \
+	which             \
+	zlib-devel
```
Comment (tomerd) [inline swift-ci/master/centos/7/Dockerfile:34]: @discogestalt this list is significantly longer than what we do for Centos8 and other distributions. Is it the bare minimum to get Swift build & qualified?

File: swift-ci/master/centos/8/Dockerfile (hunk 1 above)
Comment (discogestalt) [reply swift-ci/master/centos/8/Dockerfile]: Ok, revision coming up.

File: swift-ci/master/centos/7/Dockerfile (hunk 2 above)
Comment (discogestalt) [reply swift-ci/master/centos/7/Dockerfile:34]: For the most part, yeah - let me confirm, there may be a package or two that were there for building libcurl (when that was necessary).

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
```
[0mThe command '/bin/sh -c ln -sf /usr/bin/ninja-build /usr/bin/ninja &&   ln -sf /usr/bin/cmake3 /usr/bin/cmake &&   ln -sf /usr/lib64/libstdc++.so.6.0.?? /usr/lib64/libstdc++.so &&   ln -sf /usr/lib64/libatomic.so.1.?.? /usr/lib64/libatomic.so &&   echo "/* GNU ld script" > /usr/lib64/libgcc_s.so &&   echo "Use the shared library, but some functions are only in" >> /usr/lib64/libgcc_s.so &&   echo "the static library.  */" >> /usr/lib64/libgcc_s.so &&   echo "GROUP ( libgcc_s.so.1 -lgcc )" >> /usr/lib64/libgcc_s.so &&   sed -i -e 's/\*__block/\*__libc_block/g' /usr/include/unistd.h &&   echo -e ". /opt/rh/sclo-git25/enable\n. /opt/rh/llvm-toolset-7/enable\n. /opt/rh/devtoolset-8/enable\n" >> /home/build-user/.bashrc &&   chown -R build-user:build-user /home/build-user' returned a non-zero code: 1
```
full log - https://ci.swift.org/job/swift-docker-PR-linux/91/artifact/FAILED_swift-ci_master_centos_7.log

@discogestalt Can you look at this failure?

**Comment by 1oo7** [comment]
⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️
+1 to this PR! My company could really use this, thanks for the effort!
⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️⚜️

**Comment by discogestalt** [comment]
@shahmishal Sorry for a late response - lot's going on, and @drexin has been working on some SPM fixes related to CentOS7. He just put up PR #185 as an iteration on this one, so I'm closing this in preference to the new one.

**Pull request closed without merging**